src/features/items.ts

```diff
@@ -0,0 +1,134 @@
+import { configureStore } from "@reduxjs/toolkit";
+import { ListItem } from "../types/ListItem";
+import { Tag } from "../types/Tag";
+
+export const ADD_NEW_ITEM = "ADD_NEW_ITEM";
+export type AddNewItemActionType = {
+  type: typeof ADD_NEW_ITEM;
+  payload: ListItem;
+};
+export const addNewItem: (url: string) => AddNewItemActionType = (url) => {
+  return {
+    type: ADD_NEW_ITEM,
+    payload: {
+      id: (ItemsStore.getState().length + 1).toString(),
+      url: url,
+      date: new Date().getTime().toString(),
+      alreadyRead: false,
+      tags: [],
+    },
+  };
+};
+
+export const ASSIGN_TAG_TO_ITEM = "ASSIGN_TAG_TO_ITEM";
+export type AssignTagToItemActionType = {
+  type: typeof ASSIGN_TAG_TO_ITEM;
+  payload: ListItem;
+};
+export const assignTagToItem: (
+  tag: Tag,
+  item: ListItem
+) => AssignTagToItemActionType = (tag, item) => {
+  return {
+    type: ASSIGN_TAG_TO_ITEM,
```

Comment:
I think this logic should happen within appropriate part of reducer (in this case line 89 and below), not in action creator. While I think action creator can do things like `addNewItem()` does, in this case we're modifying incoming parameters and this does not feel good.

Imo a `AssignTagToItemActionType` should have three properties:
```
{
type: typeof ASSIGN_TAG_TO_ITEM;
tag: Tag;
item: ListItem
}
```
and reducer handles the logic. By doing that we're getting rid of genericly named `payload` property. It's completely fine if returned action has more properties than just the generic `type` & `payload`. Also, it's kinda usual to have a `type` property there, `payload` is not much self-explaining.